### PR TITLE
Add handling for grant/revoke all tables in schema

### PR DESCRIPTION
**Pull request by mtuncer**
Preliminaries for supporting ```grant/revoke XXX on all tables in schema ...```

### CODE REVIEW

File: src/backend/distributed/executor/multi_utility.c [hunk 1]
```diff
@@ -3823,6 +3834,79 @@ PlanGrantStmt(GrantStmt *grantStmt)
...
+	grantAllTablesOnSchenaCommand = (grantStmt->targtype == ACL_TARGET_ALL_IN_SCHEMA &&
+									 grantStmt->objtype == RELATION_OBJECT_TYPE);
+
+	/* we are only interested in table level grants */
+	if (!grantOnTableCommand && !grantAllTablesOnSchenaCommand)
+	{
+		return NIL;
+	}
+
+	if (grantAllTablesOnSchenaCommand)
```
Comment (marcocitus) [inline src/backend/distributed/executor/multi_utility.c]: Schena -> Schema

File: src/backend/distributed/executor/multi_utility.c [hunk 2]
```diff
@@ -3823,6 +3834,79 @@ PlanGrantStmt(GrantStmt *grantStmt)
...
+	{
+		List *distTableOidList = DistTableOidList();
+		ListCell *distributedTablOidCell = NULL;
```
Comment (marcocitus) [inline src/backend/distributed/executor/multi_utility.c]: Tabl -> Table

File: src/backend/distributed/executor/multi_utility.c [hunk 3]
```diff
@@ -3823,6 +3834,79 @@ PlanGrantStmt(GrantStmt *grantStmt)
...
+			bool missing_ok = false;
+			Oid nspId = get_namespace_oid(nspname, missing_ok);
+			Assert(nspId != InvalidOid);
+			namespaceOidList = list_append_unique_oid(namespaceOidList, nspId);
+		}
+
+		foreach(distributedTablOidCell, distTableOidList)
+		{
+			Oid relid = lfirst_oid(distributedTablOidCell);
+			Oid nspId = get_rel_namespace(relid);
```
Comment (marcocitus) [inline src/backend/distributed/executor/multi_utility.c]: we should be using wordy variable names (e.g. `namespaceOid`)

File: src/backend/distributed/executor/multi_utility.c [hunk 4]
```diff
@@ -3727,7 +3736,13 @@ PlanGrantStmt(GrantStmt *grantStmt)
 			}
 			isFirst = false;
 
-			Assert(priv->cols == NIL);
+			if (priv->cols != NIL)
+			{
+				ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
```
Comment (marcocitus) [inline src/backend/distributed/executor/multi_utility.c]: have we established that we're dealing with distributed tables at this point?

Comment (mtuncer) [reply src/backend/distributed/executor/multi_utility.c]: correct, only distributed tables are returned from ```CollectGrantTableList```

File: src/backend/distributed/executor/multi_utility.c (hunk 3 above)
Comment (mtuncer) [reply src/backend/distributed/executor/multi_utility.c]: done

File: src/backend/distributed/executor/multi_utility.c [hunk 5]
```diff
@@ -3823,6 +3834,78 @@ PlanGrantStmt(GrantStmt *grantStmt)
...
+ *  Oids from grant statement.
+ *  Grant statement may appear in two forms
+ *  1 - grant on table:
+ *      each distributed table oid in grant object list is added to returned list.
+ *  2 - grant all tables in schema:
+ *     Collect namespace oid list from grant statement
+ *     Add each distributed table oid in the target namespace list to the returned list.
+ */
+static List *
+CollectGrantTableList(GrantStmt *grantStmt)
```
[the remark this answers is not in the document]
Comment (mtuncer) [reply src/backend/distributed/executor/multi_utility.c]: ok

**Pull request merged**